scripts/build.py

```diff
@@ -105,21 +102,38 @@ def copy_files_source_to_target(source, target):
             shutil.copyfile(source_path, target_path)
 
 
-def _build_files():
-    ensure_directory_exists(OUT_DIR)
-    shutil.rmtree(OUT_DIR)
-    process_third_party_libs()
+def build_files(source, target, ignore=None):
+    """Minifies all css and js files, and removes whitespace from html in source
+    directory and copies it to target, ignoring paths/files mentioned in ignore.
+    Copies files in ignore to target without any changes.
+    """
+    print 'Processing %s' % os.path.join(os.getcwd(), source)
+    print 'Generating into %s' % os.path.join(os.getcwd(), target)
+    ensure_directory_exists(target)
+    shutil.rmtree(target)
 
-    for root, dirs, files in os.walk(os.path.join(os.getcwd(), HEAD_DIR)):
+    for root, dirs, files in os.walk(os.path.join(os.getcwd(), source)):
         for directory in dirs:
             print 'Processing %s' % os.path.join(root, directory)
         for filename in files:
             source_path = os.path.join(root, filename)
-            if source_path.find(OUT_DIR) > 0:
+            if source_path.find(target) > 0:
                 continue
-            if source_path.find(HEAD_DIR) == -1:
+            if source_path.find(source) == -1:
                 continue
-            target_path = source_path.replace(HEAD_DIR, OUT_DIR)
+            target_path = source_path.replace(source, target)
+
+            only_copy_file = False
+            if (any(source_path.find(p) > 0 for p in ignore) or
+                    any(source_path.find(p) > 0 for p in ['.json']) or
+                    not any(source_path.find(p) > 0 for p in ['.html', '.css', '.js'])):
```

Comment:
I'm thinking ``['.html', '.css', '.js']`` and ``['.json']`` should be file-level constants to provide better context for their values. WDYT?